File: tools/benchmark/nori_clone_runner.go
package main

import (
	"bufio"
	"log"
	"os"
	"strconv"
	"time"

	nori "github.com/jeongukjae/nori-clone/nori/go"
)

func main() {
	tokenizer, err := nori.New("./dictionary/latest-dictionary.nori", "./dictionary/latest-userdict.txt")

	if err != nil {
		log.Fatal(err)
	}
	defer tokenizer.Free()

	file, err := os.Open(os.Args[1])
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	var lines []string
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}
	n, err := strconv.Atoi(os.Args[2])
	if err != nil {
		log.Fatal(err)
	}
	lines = lines[:n]

	start := time.Now()
	for _, line := range lines {
		tokenizer.Tokenize(line)
	}
	elapsed := time.Since(start)
	log.Printf("Elapsed %s", elapsed)
}

File: tools/benchmark/nori_clone_runner.py
import time
from sys import argv, stdin

import nori

tokenizer = nori.NoriTokenizer()
tokenizer.load_prebuilt_dictionary("./dictionary/latest-dictionary.nori")
tokenizer.load_user_dictionary("./dictionary/latest-userdict.txt")

def run_with_iterator(f):
    for line in f:
        tokenizer.tokenize(line)


if len(argv) != 1:
    # read all lines from the input file
    with open(argv[1]) as f:
        lines = f.readlines()[:int(argv[2])]

    start_time = time.time()
    run_with_iterator(lines)
else:
    start_time = time.time()
    run_with_iterator(stdin)

end_time = time.time()
time_diff = (end_time - start_time)
print(int(time_diff * 1000))
